File: src/main/groovy/net/zomis/brainf/model/ast/Token.java
package net.zomis.brainf.model.ast;

public class Token {

    private TokenInfo info;

    public Token(TokenInfo tokenInfo) {
        this.info = new TokenInfo(tokenInfo);
    }

    public TokenInfo getInfo() {
        return new TokenInfo(info);
    }

    @Override
    public String toString() {
        return "Token{" +
                "info=" + info +
                '}';
    }

}

File: brainduck-kotlin/src/commonMain/kotlin/net/zomis/brainf/model/ast/Token.kt
package net.zomis.brainf.model.ast;

open class Token(tokenInfo: TokenInfo) {
    private val _info = tokenInfo.copy()
    val info: TokenInfo get() = _info.copy()

    override fun toString(): String {
        return "Token{" +
                "info=" + info +
                '}';
    }

}
